lib/endpoints/networks.js

```diff
@@ -263,12 +264,36 @@ function listNetworks(req, res, next) {
         return translateNetwork(req, pool);
     });
 
+    function resolveSuffix(n, cb) {
+    var account = req.account.uuid;
+    var nets = [n.id];
+    req.sdc.cns.getSuffixesForVM(account, nets, function (err, suf) {
+            n.suffixes = suf.suffixes;
+            return cb(err, n);
+        });
+    }
+
+    if (req.sdc.cns && req.account.triton_cns_enabled === 'true') {
+        vasync.forEachParallel({
+            'func': resolveSuffix,
+            'inputs': networks
+        }, function (err, results) {
+            if (err) {
+                req.log.warn('failed to retrieve DNS suffixes from CNS' +
```

Comment:
Do we fail the whole operation if getSuffixesForVM() fail?  I don't see a callback or anything.

Reply:
The code is written to attempt to return the network(s) without the suffix information, but if there's an error (like cns is down) cloudapi will throw an unhandled internal error and the result won't go back to the client (I'm not 100% sure why, because I intended to just return it without the suffix info, but this is probably much deeper in cloudapi).

The same thing happens with vms if you try to fetch vm details while cns is unavailable you'll get an error rather than the vm info without the dns_names. So if you have CNS set up, but it's unhealthy for whatever reason, cloudapi is going to have a hard time in several areas. This maybe could be improved at some point, but given that we have similar behavior for vms and networks, and the vm behavior is pre-existing, I think this is probably a satisfactory situation.